src/metrics.rs

```diff
@@ -55,8 +58,10 @@ pub async fn metrics_handler() -> String {
 /// Gather relevant prometheus metrics on all incoming requests
 pub fn record_request_metrics(request: &Request<Body>, _span: &Span) {
     // Increment request counter
+    let http_method = &request.method().to_string().to_ascii_uppercase();
+    let request_path = &request.uri().to_string();
```

Comment:
Maybe just the path component?

Reply:
Looks like the URI method only gives us the last section of the path anyway. E.g. `curl localhost:8080/v1/count -X post` gives us a metric that looks like `incoming_requests{http_method="POST",path="/count"} 1`. Maybe due to where the tracing layer is in the router stack at the moment?

Reply:
Does it give `/count` or `/v1/count`? Either way, I think `request.uri().path().to_string()` should work

Reply:
I've moved the middleware layer to the first in the stack so that it now gives me the full path e.g. `/v1/count`